### PR TITLE
add unique

### PR DESCRIPTION
**Pull request by bilal-aamer**
Close https://github.com/unifyai/ivy/issues/21866

### CODE REVIEW

**Comment by jshepherd01** [comment]
Your test seems to be failing when I run it
```py
ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_manipulation.py:977: in <module>
    def test_unique(*, dtype_x_axis, none_axis, test_flags, backend_fw, fn_name, on_device):
ivy_tests/test_ivy/helpers/testing_helpers.py:363: in test_wrapper
    callable_fn, fn_name, fn_mod = _import_fn(fn_tree)
ivy_tests/test_ivy/helpers/testing_helpers.py:157: in _import_fn
    callable_fn = mod.__dict__[fn_name]
E   KeyError: 'unique'
```
because there doesn't seem to be a `unique` function in `ivy/experimental/manipulation.py`.

Also, could you please link your issue to the relevant ToDo list? It massively helps with bookkeeping to know what everyone's working on

**Comment by bilal-aamer** [comment]
Hey @jshepherd01, thanks for the input will be putting a commit and linking ToDo issue.

**Comment by jshepherd01** [comment]
The tests are still failing for me:
```py
FAILED ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_manipulation.py::test_unique[cpu-numpy-False-False] - ivy.utils.exceptions.IvyBackendException: numpy: unique: numpy: unique: numpy: unique: numpy: unique: numpy: unique: numpy: unique: numpy: unique: numpy: unique: numpy: unique: numpy: unique: numpy: uniqu...
FAILED ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_manipulation.py::test_unique[cpu-jax-False-False] - hypothesis.errors.Flaky: Inconsistent test results! Test case was Conclusion(status=Status.INTERESTING, interesting_origin=(<class 'ivy.utils.exceptions.IvyBackendException'>, '/home/joe/dev/ivy/ivy/utils...
FAILED ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_manipulation.py::test_unique[cpu-tensorflow-False-False] - ivy.utils.exceptions.IvyBackendException: tensorflow: unique: tensorflow: unique: tensorflow: unique: tensorflow: unique: tensorflow: unique: tensorflow: unique: tensorflow: unique: tensorflow: unique: te...
FAILED ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_manipulation.py::test_unique[cpu-paddle-False-False] - hypothesis.errors.Flaky: Inconsistent test results! Test case was Conclusion(status=Status.INTERESTING, interesting_origin=(<class 'ivy.utils.exceptions.IvyBackendException'>, '/home/joe/dev/ivy/ivy/utils...
```
I'm also seeing a failed linting test which can probably be fixed by running the [pre-commit hooks](https://unify.ai/docs/ivy/overview/contributing/setting_up.html#pre-commit).

Also, I think your issue still needs to be linked to the ToDo list issue properly (following the instructions [here](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues)), and based on that issue it seems like you need to add `unique` to the paddle frontend rather than the `ivy.Array` class (see [here](https://unify.ai/docs/ivy/overview/contributing/open_tasks.html#frontend-apis) for more information about that).

**Comment by jshepherd01** [comment]
I'm still seeing failed tests, do you get the same thing?
image

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_manipulation.py [hunk 1]
```diff
@@ -674,3 +674,33 @@ def test_paddle_rot90(
...
+    dtype_x_axis=helpers.dtype_values_axis(
+        available_dtypes=helpers.get_dtypes("numeric"),
+        min_num_dims=1,
+        min_dim_size=2,
+        force_int_axis=True,
+        valid_axis=True,
+    ),
+    none_axis=st.booleans(),
+    test_with_out=st.just(False),
+    test_gradients=st.just(False),
```
Comment (jshepherd01) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_manipulation.py:691]: The tests currently fail with `hypothesis.errors.InvalidArgument: test_unique() got an unexpected keyword argument 'test_gradients', from 'test_gradients=just(False)'`

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_manipulation.py [hunk 2]
```diff
@@ -674,3 +674,33 @@ def test_paddle_rot90(
...
+    ground_truth_backend="torch",
+)
+def test_unique(*, dtype_x_axis, none_axis, test_flags, backend_fw, fn_name, on_device):
```
Comment (jshepherd01) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_manipulation.py:694]: Frontend test functions should be named `test_frontend_funcname` (i.e. `test_paddle_unique`)

**Review by jshepherd01 — CHANGES_REQUESTED** [review]

**Comment by bilal-aamer** [comment]
@jshepherd01 made the required changes

**Comment by bilal-aamer** [comment]
@jshepherd01 made the required changes

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_manipulation.py [hunk 3]
```diff
@@ -674,3 +674,32 @@ def test_paddle_rot90(
...
+    dtype_x_axis=helpers.dtype_values_axis(
+        available_dtypes=helpers.get_dtypes("numeric"),
+        min_num_dims=1,
+        min_dim_size=2,
+        force_int_axis=True,
+        valid_axis=True,
+    ),
+    none_axis=st.booleans(),
+    test_with_out=st.just(False),
+    ground_truth_backend="torch",
```
Comment (jshepherd01) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_manipulation.py:691]: Tests now throw the error `hypothesis.errors.InvalidArgument: test_paddle_unique() got an unexpected keyword argument 'ground_truth_backend'`.

**Review by jshepherd01 — CHANGES_REQUESTED** [review]
Tests are still failing for me, are they passing on your machine?

**Comment by bilal-aamer** [comment]
> Tests are still failing for me, are they passing on your machine?

Can't see the tests but most passed successfully

**Comment by jshepherd01** [comment]
If you can't see the tests then that's a problem in itself. You can run them locally from the `ivy` directory using `python3 -m pytest path/to/test/file.py::test_function_name -v` and see the results for yourself.

Also, I can still see some linting errors resulting from your code, please double check that you've installed the [pre-commit hooks](https://unify.ai/docs/ivy/overview/contributing/setting_up.html#pre-commit) correctly and that they're running every time you use `git commit`.

Thanks

**Review by juliagsy — CHANGES_REQUESTED** [review]
Hey! Thanks for the hard work! Just a quick one, would you mind quickly resolving the conflicts in this PR first? I'll try to check the tests after that soonest possible, thanks!

**Review by juliagsy — COMMENTED** [review]
Hey! I see that there are a lot of additions after resolving the merge conflicts, I believe these function are either already existing in another file, or a duplication of the same file, could you please cross check with the latest code in the master branch, and keep only the changes related to you frontend function in your issue? Thanks!

**Comment by juliagsy** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Comment by bilal-aamer** [comment]
Hey @juliagsy

**Review by juliagsy — COMMENTED** [review]
Hey! I couldn't see the refactor from the PR diff unfortunately, it's still showing a load of function additions which are not related to your chosen function. Could you please either push your changes to the PR so that it's up-to-date or could you please refactor them again? thanks!

**Comment by juliagsy** [comment]
This PR has been closed because it has been stale for more than 7 days with no activity. Please feel free to reopen the PR yourself if you'd like to continue, thanks!

**Comment by bilal-aamer** [comment]
@juliagsy I'd like to continue

**Pull request closed without merging**